src/main/java/org/opensearch/plugin/insights/core/service/TopQueriesService.java

```diff
@@ -93,6 +96,8 @@ public class TopQueriesService {
      */
     private QueryInsightsExporter exporter;
 
+    private MinMaxHeapQueryGrouper minMaxHeapQueryGrouper;
```

Comment:
Again we should use interface now, MinMaxHeapQueryGrouper -> QueryGrouper

and rename minMaxHeapQueryGrouper -> queryGrouper